### PR TITLE
Add CC Functionality

### PR DESCRIPTION
**Pull request by Bentechy66**
Currently you can only create a CC, adding / removing is not yet implemented.

### CODE REVIEW

File: config.py [hunk 1]
```diff
@@ -14,14 +14,14 @@
...
 
 # List of specific channels
-welcome_channel = "0" #TODO: Channel where the bot says it's gone online.
-game_log = "1" #TODO
-bot_spam = "2" #TODO
-story_time = "3" #TODO
+welcome_channel = 456880220486631424 # Channel where the bot says it's gone online.
+game_log = 1 #TODO
+bot_spam = 456880259636396034
+story_time = 3 #TODO
```
[the remark this answers is not in the document]
Comment (Bentechy66) [reply config.py]: Why bother

File: config.py [hunk 2]
```diff
@@ -1,5 +1,5 @@
 # Discord API / Login settings
-TOKEN = 'put_token_here'
+TOKEN = 'NDAxNzE2MzgwODAzNTMwNzUy.DgWohA.lhaTqFX6aFR9a6usQxnlSRCi2mM'
```
[the remark this answers is not in the document]
Comment (Bentechy66) [reply config.py]: Oops

File: conspiracy_channels/main.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+            except Exception as e:
+                # Category was invalid, create a new one now
+                print(str(e))
+                pass
+            with open("conspiracy_channels/cc_data.json", 'w') as cc_data:
+                json.dump(data, cc_data)
+
+
+def setup(bot):
+    bot.add_cog(ConspiracyCog(bot))
```
[the remark this answers is not in the document]
Comment (Bentechy66) [reply conspiracy_channels/main.py]: It's not part of the framework it's part of a better one

File: requirements.txt [hunk 4]
```diff
@@ -1,2 +1,3 @@
 discord.py
 emoji
+https://github.com/Rapptz/discord.py/archive/rewrite.zip
```
[the remark this answers is not in the document]
Comment (Bentechy66) [reply requirements.txt]: A newer version of discordpy

File: config.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Randium) [reply config.py]: 'cause organization 'n' stuff.
It's three seconds of work.

File: conspiracy_channels/main.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (Randium) [reply conspiracy_channels/main.py]: Great that you're coming with that idea NOW, now that I've written over 2000 lines of code after having agreed upon an idea.

File: config.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (romangraef) [reply config.py]: nah, not needed. those values are the ones from the testing server, so all devs prolly use them

File: conspiracy_channels/cc_data.json [hunk 5]
```diff
@@ -0,0 +1 @@
+{"category_id": 457264810363584513, "channel_num": 0}
```
Comment (romangraef) [inline conspiracy_channels/cc_data.json]: why commit this? but ok

File: conspiracy_channels/main.py [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+        message = creation_messages.cc_intro(member_ids) # This is the message that will be sent to users once the channel is created
+
+
+        #Role objects (based on ID)
+        guild = self.bot.get_channel(bot_spam).guild
+        roles = guild.roles
+        game_master_role = discord.utils.find(lambda r: r.id == game_master, roles)
+        participant_role = discord.utils.find(lambda r: r.id == participant, roles)
+        dead_participant_role = discord.utils.find(lambda r: r.id == dead_participant, roles)
+        frozen_participant_role = discord.utils.find(lambda r: r.id == frozen_participant, roles)
```
Comment (romangraef) [inline conspiracy_channels/main.py]: maybe put those into an extra method?

File: conspiracy_channels/main.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (romangraef) [reply conspiracy_channels/main.py]: yeah "agreed"... meaning everyone else was just to tired of fighting with you

File: main.py [hunk 7]
```diff
@@ -1,36 +1,43 @@
...
+@bot.event
 async def on_ready():
     print('Logged in as')
-    print(client.user.name)
-    print(client.user.id)
+    print(bot.user.name)
+    print(bot.user.id)
     print('------')
-    
-    await client.send_message(client.get_channel(welcome_channel),'Beep boop! I just went online!')
```
Comment (romangraef) [inline main.py]: why delete this? it was kinda usefull and not really annoying if you just muted that channel

File: management/admin.py [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+        """Reloads a module."""
+        try:
+            self.bot.unload_extension(module)
+            self.bot.load_extension(module)
+        except Exception as e:
+            await ctx.channel.send('\N{PISTOL}')
+            await ctx.channel.send('{}: {}'.format(type(e).__name__, e))
+        else:
+            await ctx.channel.send('\N{OK HAND SIGN}')
+
```
Comment (romangraef) [inline management/admin.py]: make all that GM only please. see https://github.com/werewolves-devs/Werewolf_Bot/blob/5064b641813bc9971c541aba0ddca41f4fe6b757/utils.py#L40

File: requirements.txt (hunk 4 above)
[the remark this answers is not in the document]
Comment (romangraef) [reply requirements.txt]: and we need it for reaction handling. the old version only supports this to some *very* limited extent

File: main.py (hunk 7 above)
Comment (Bentechy66) [reply main.py]: I moved it

File: config.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Bentechy66) [reply config.py]: Yea, that's why I left them in

File: conspiracy_channels/main.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (Bentechy66) [reply conspiracy_channels/main.py]: This is a far easier and better framework to work with, as far as I can see.

File: requirements.txt (hunk 4 above)
[the remark this answers is not in the document]
Comment (Bentechy66) [reply requirements.txt]: Plus doesn't support categories at all

**Pull request merged**